temporian/core/operators/propagate.py

```diff
@@ -46,18 +46,18 @@ def __init__(
         if len(to.features()) == 0:
             raise ValueError("to contains no features")
 
-        self._added_index = [k.name() for k in to.features()]
+        self._added_index = {k.name(): k.dtype() for k in to.features()}
 
         overlap_features = set(self._added_index).intersection(
-            event.sampling().index()
+            set(event.sampling().index())
         )
         if len(overlap_features) > 0:
             raise ValueError(
                 "to contains feature names already present in the"
                 f" index: {list(overlap_features)}"
             )
 
-        new_index = event.sampling().index() + self._added_index
+        new_index = {**event.sampling().index(), **self._added_index}

```

Comment:
Afterthoughts: I would implement the index list immediately. This index dictionary add a lot of changes. Note: You can also create a method sampling().index_keys() to minimize the change.

Reply:
okay. I was thinking of doing something like this:

```
IndexDtypes = Union[dtype_lib.INT32, dtype_lib.INT64, dtype_lib.STRING]


class IndexLevel(NamedTuple):
    name: str
    dtype: IndexDtypes


class Sampling(object):
    def __init__(
        self,
        index: List[Tuple[str, IndexLevel]],
        ...
     ):
     self._index = [IndexLevel(name, dtype) for name, dtype in index] # convert regular tuples to IndexLevel
     ...
``` 

I prefer the user being able to specify the `index` with native python objects rather than having to call `IndexLevel`, which seems too verbose. What do you think?